eng/pipelines/templates/Build_and_UnitTest.yml

```diff
@@ -113,7 +113,7 @@ steps:
   inputs:
     solution: "src\\NuGet.Clients\\NuGet.CommandLine\\NuGet.CommandLine.csproj"
     configuration: "$(BuildConfiguration)"
-    msbuildArguments: "/t:ILMergeNuGetExe /p:ExpectedLocalizedArtifactCount=$(LocalizedLanguageCount) /bl:$(Build.StagingDirectory)\\binlog\\04.ILMergeNuGetExe.binlog"
+    msbuildArguments: "/t:Build,ILMergeNuGetExe /bl:$(Build.StagingDirectory)\\binlog\\04.ILMergeNuGetExe.binlog"
```

Comment:
Should we have separate calls for the build and ilmerge target. I know that'll affect perf negatively, but might be easier to reason about whenever we encounter failures.

Curious what the folks that have done CI work think. cc @zivkan

Reply:
I'm good doing a single or a separate pipeline step. @zivkan What do you think?

Reply:
This is the commit that made ilmerge happen as a separate step in the first place: https://github.com/NuGet/NuGet.Client/commit/a746cd6a01dfd22187375ee9c5a0d9b98676020a

It appears to have been done because of loc issues.  If that's no longer a problem, then doing ilmerge in the normal build would be good (less customization the better).

Reply:
I think we still need to build passes for

1. Build first binaries with English resources
2. Run localize step and produce all additional localized .resources.dll assemblies
3. Build final nuget.exe with embedded localized assemblies

Reply:
Ok, I just took a closer look at this.

Why is the `Build` target needed in this task, when it already ran on line 97's build task?

Reply:
Because that's the second build. This is the updated list:

1. Build first binaries with English resources (line 97)
2. Run localize step and produce all additional localized .resources.dll assemblies (line 104)
3. Build final nuget.exe with embedded localized assemblies (line 116)

Reply:
You forgot step 4, ilmerge all build assemblies into a single file.

I had a second look, and I see now that we're using the compiler's "embed resource", not ilmerge, for the satellite assemblies.

Nikolche's original comment is about splitting steps 3 (second build) & 4 (ilmerge) into separate steps, meaning I completely misunderstood the conversation last week.  Personally, I don't mind either way.